crates/hir_ty/src/lower.rs

```diff
@@ -287,15 +295,50 @@ impl<'a> TyLoweringContext<'a> {
                     }
                 }
             }
-            mt @ TypeRef::Macro(_) => {
-                if let Some(module_id) = self.resolver.module() {
-                    match expand_macro_type(self.db.upcast(), module_id, mt) {
-                        Some(type_ref) => self.lower_ty(&type_ref),
-                        None => TyKind::Error.intern(&Interner),
+            TypeRef::Macro(macro_call) => {
+                let (expander, recursion_start) = match self.expander.borrow_mut() {
```

Comment:
This doesn't look like it needs a `match`

Reply:
I did that because I could then move the `RefMut` around so it would "automatically drop" when `None` is returned.

Reply:
Or I could just use a block.